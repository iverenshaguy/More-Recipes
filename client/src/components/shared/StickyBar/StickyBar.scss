@import '../../../assets/scss/modules';

.sticky-bar-wrapper {
  bottom: 0;
  color: $white;
  position: fixed;
  width: 100%;
}

.sticky-bar {
  background-color: $white;
  // bottom: 0;
  color: $light-green;
  // height: 6rem;
  padding: 1rem 0;
  width: 100%;

  i,
  span {
    color: $light-green;
  }
}

.home {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.user-profile {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.favorite-recipes {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.user-recipe {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.top-recipes {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.search-recipes {
  .sticky-bar {
    div {
      padding-right: 2.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 1em;
      }
    }
  }
}

.view-recipe {
  .sticky-bar {
    div {
      padding-right: 1.5em;

      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 0.5em;
      }
    }
  }
}

@media only screen and (max-width: 768px) and (min-width: 576px) {
  .home {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .user-profile {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .favorite-recipes {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .user-recipe {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .top-recipes {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .search-recipes {
    .sticky-bar {
      div {
        padding-right: 7em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 1em;
        }
      }
    }
  }

  .view-recipe {
    .sticky-bar {
      div {
        padding-right: 4.2em;

        &:first-child {
          padding-left: 0;
        }

        &:last-child {
          padding-right: 0.2em;
        }
      }
    }
  }
}

@media only screen and (max-width: 575px) and (min-width: 450px) {
  .search-recipes {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .home {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .user-profile {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .favorite-recipes {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .user-recipe {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .top-recipes {
    .sticky-bar {
      div {
        padding-right: 5em;

        &:first-child {
          padding-left: 2em;
        }

        &:last-child {
          padding-right: 2em;
        }
      }
    }
  }

  .view-recipe {
    .sticky-bar {
      div {
        padding-right: 2em;

        &:first-child {
          padding-left: 0.5em;
        }

        &:last-child {
          padding-right: 0.5em;
        }
      }
    }
  }
}

#result-search-form-div {
  background-color: $transparent-light;

  #search-form {
    #search {
      border: 1px solid $light-green !important;
    }

    .input-group-btn {
      button {
        background-color: $light-green !important;
        border-color: $light-green !important;
        color: $white !important;
      }
    }
  }
}