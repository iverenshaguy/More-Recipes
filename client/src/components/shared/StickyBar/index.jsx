import React from 'react';
import { connect } from 'react-redux';
import FontAwesome from 'react-fontawesome';
import { Form, InputGroup, Label, Input, InputGroupButton, Button } from 'reactstrap';
import PropTypes from 'prop-types';
import { componentActions } from '../../../store/components';
import './StickyBar.scss';

const { toggleModal } = componentActions;

const StickyBar = ({ currentLocation, dispatch }) => {
  if (currentLocation === 'auth') {
    return '';
  }

  return (
    <div className={currentLocation}>
      <div className="sticky-bar-wrapper text-center d-block d-sm-block d-md-none">
        <div className="container-fluid">
          <div className="row" id="result-search-form-div">
            <div className="col-12 w-100 py-3">
              <Form id="search-form">
                <InputGroup>
                  <Label className="sr-only" for="search">
                    Search for Recipe
                  </Label>
                  <Input type="text" id="search" placeholder="Search for More Recipes" />
                  <InputGroupButton className="input-group-btn">
                    <Button type="submit">Submit</Button>
                  </InputGroupButton>
                </InputGroup>
              </Form>
            </div>
          </div>
        </div>
        <div className="sticky-bar text-center">
          {currentLocation === 'user-profile' && (
            <a className="nav-link d-inline">
              <i className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods" />
            </a>
          )}
          {currentLocation !== 'user-profile' && (
            <a className="nav-link d-inline" data-toggle="modal">
              <FontAwesome
                name="share-alt"
                id="social-modal-icon"
                size="lg"
                onClick={() => dispatch(toggleModal('social'))}
              />
            </a>
          )}
          <a className="nav-link d-inline" title="New Recipe">
            <FontAwesome
              name="plus"
              id="add-edit-modal-icon"
              size="lg"
              onClick={() => dispatch(toggleModal('addRecipe'))}
            />
          </a>
          {currentLocation === 'view-recipe' && (
            <a className="nav-link d-inline favorite" title="Favorite">
              <FontAwesome name="heart-o" size="lg" />
            </a>
          )}
          {currentLocation === 'view-recipe' && (
            <a className="nav-link d-inline upvote" title="Upvote">
              <FontAwesome name="thumbs-o-up" size="lg" />
            </a>
          )}
          {currentLocation === 'view-recipe' && (
            <a className="nav-link d-inline downvote" title="Downvote">
              <FontAwesome name="thumbs-o-down" size="lg" />
            </a>
          )}
          <a className="nav-link d-inline">
            <FontAwesome name="search" size="lg" className="search-icon" />
          </a>
          {currentLocation !== 'view-profile' && (
            <a className="nav-link d-inline">
              <FontAwesome name="heart" size="lg" />
            </a>
          )}
        </div>
      </div>
    </div>
  );
};

StickyBar.propTypes = {
  dispatch: PropTypes.func.isRequired,
  currentLocation: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  currentLocation: state.location.current,
});

export { StickyBar as StickyBarComponent };

export default connect(mapStateToProps)(StickyBar);
