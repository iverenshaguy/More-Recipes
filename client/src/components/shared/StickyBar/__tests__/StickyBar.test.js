import React from 'react';
import StickyBar from '../index';

describe('StickyBar', () => {
  const dispatchMock = jest.fn();

  afterAll(() => {
    jest.clearAllMocks();
  });

  it("renders oven icon when type 'user-profile' is passed", () => {
    const wrapper = mount(<StickyBar dispatch={dispatchMock} currentLocation="user-profile" />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.flaticon-oven-kitchen-tool-for-cooking-foods')).toBeTruthy();
    expect(wrapper.find('.fa-plus')).toBeTruthy();
    expect(wrapper.find('.fa-search')).toBeTruthy();
    expect(wrapper.find('.fa-heart')).toBeTruthy();

    wrapper.unmount();
  });

  it("renders no sticky bar when type 'login' is passed", () => {
    const wrapper = mount(<StickyBar dispatch={dispatchMock} currentLocation="auth" />);

    expect(toJson(wrapper)).toMatchSnapshot();

    expect(wrapper.instance()).toBeNull();

    wrapper.unmount();
  });

  it("renders no sticky bar when type 'signup' is passed", () => {
    const wrapper = mount(<StickyBar dispatch={dispatchMock} currentLocation="auth" />);

    expect(toJson(wrapper)).toMatchSnapshot();

    expect(wrapper.instance()).toBeNull();

    wrapper.unmount();
  });

  it("renders share icon when type 'home' is passed", () => {
    const wrapper = mount(<StickyBar dispatch={dispatchMock} currentLocation="home" />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.fa-share-alt')).toBeTruthy();
    expect(wrapper.find('.fa-plus')).toBeTruthy();
    expect(wrapper.find('.fa-search')).toBeTruthy();
    expect(wrapper.find('.fa-heart')).toBeTruthy();

    wrapper.unmount();
  });

  it("renders heart-o, thumbs-up-o, thumbs-down-o icons and does not render heart icon when type 'view-recipe' is passed", () => {
    const wrapper = mount(<StickyBar dispatch={dispatchMock} currentLocation="view-recipe" />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.fa-share-alt')).toBeTruthy();
    expect(wrapper.find('.fa-plus')).toBeTruthy();
    expect(wrapper.find('.fa-heart-o')).toBeTruthy();
    expect(wrapper.find('.fa-thumbs-up-o')).toBeTruthy();
    expect(wrapper.find('.fa-thumbs-down-o')).toBeTruthy();
    expect(wrapper.find('.fa-search')).toBeTruthy();

    wrapper.unmount();
  });
});
