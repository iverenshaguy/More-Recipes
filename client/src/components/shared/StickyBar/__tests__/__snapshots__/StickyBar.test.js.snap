// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyBar renders heart-o, thumbs-up-o, thumbs-down-o icons and does not render heart icon when type 'view-recipe' is passed 1`] = `
<StickyBar
  currentLocation="view-recipe"
  dispatch={[Function]}
>
  <div
    className="view-recipe"
  >
    <div
      className="sticky-bar-wrapper text-center d-block d-sm-block d-md-none"
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
          id="result-search-form-div"
        >
          <div
            className="col-12 w-100 py-3"
          >
            <Form
              id="search-form"
              tag="form"
            >
              <form
                className=""
                id="search-form"
              >
                <InputGroup
                  tag="div"
                >
                  <div
                    className="input-group"
                  >
                    <Label
                      className="sr-only"
                      for="search"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <label
                        className="sr-only  form-control-label"
                        htmlFor="search"
                      >
                        Search for Recipe
                      </label>
                    </Label>
                    <Input
                      id="search"
                      placeholder="Search for More Recipes"
                      tag="p"
                      type="text"
                    >
                      <input
                        className="form-control"
                        id="search"
                        placeholder="Search for More Recipes"
                        type="text"
                      />
                    </Input>
                    <InputGroupButton
                      className="input-group-btn"
                      tag="div"
                    >
                      <div
                        className="input-group-btn input-group-btn"
                      >
                        <Button
                          color="secondary"
                          tag="button"
                          type="submit"
                        >
                          <button
                            className="btn btn-secondary"
                            onClick={[Function]}
                            type="submit"
                          >
                            Submit
                          </button>
                        </Button>
                      </div>
                    </InputGroupButton>
                  </div>
                </InputGroup>
              </form>
            </Form>
          </div>
        </div>
      </div>
      <div
        className="sticky-bar text-center"
      >
        <a
          className="nav-link d-inline"
          data-toggle="modal"
        >
          <FontAwesome
            id="social-modal-icon"
            name="share-alt"
            onClick={[Function]}
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-share-alt fa-lg"
              id="social-modal-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
          title="New Recipe"
        >
          <FontAwesome
            id="add-edit-modal-icon"
            name="plus"
            onClick={[Function]}
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-plus fa-lg"
              id="add-edit-modal-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline favorite"
          title="Favorite"
        >
          <FontAwesome
            name="heart-o"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-heart-o fa-lg"
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline upvote"
          title="Upvote"
        >
          <FontAwesome
            name="thumbs-o-up"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-thumbs-o-up fa-lg"
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline downvote"
          title="Downvote"
        >
          <FontAwesome
            name="thumbs-o-down"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-thumbs-o-down fa-lg"
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
        >
          <FontAwesome
            className="search-icon"
            name="search"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-search fa-lg search-icon"
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
        >
          <FontAwesome
            name="heart"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-heart fa-lg"
            />
          </FontAwesome>
        </a>
      </div>
    </div>
  </div>
</StickyBar>
`;

exports[`StickyBar renders no sticky bar when type 'login' is passed 1`] = `
<StickyBar
  currentLocation="auth"
  dispatch={[Function]}
/>
`;

exports[`StickyBar renders no sticky bar when type 'signup' is passed 1`] = `
<StickyBar
  currentLocation="auth"
  dispatch={[Function]}
/>
`;

exports[`StickyBar renders oven icon when type 'user-profile' is passed 1`] = `
<StickyBar
  currentLocation="user-profile"
  dispatch={[Function]}
>
  <div
    className="user-profile"
  >
    <div
      className="sticky-bar-wrapper text-center d-block d-sm-block d-md-none"
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
          id="result-search-form-div"
        >
          <div
            className="col-12 w-100 py-3"
          >
            <Form
              id="search-form"
              tag="form"
            >
              <form
                className=""
                id="search-form"
              >
                <InputGroup
                  tag="div"
                >
                  <div
                    className="input-group"
                  >
                    <Label
                      className="sr-only"
                      for="search"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <label
                        className="sr-only  form-control-label"
                        htmlFor="search"
                      >
                        Search for Recipe
                      </label>
                    </Label>
                    <Input
                      id="search"
                      placeholder="Search for More Recipes"
                      tag="p"
                      type="text"
                    >
                      <input
                        className="form-control"
                        id="search"
                        placeholder="Search for More Recipes"
                        type="text"
                      />
                    </Input>
                    <InputGroupButton
                      className="input-group-btn"
                      tag="div"
                    >
                      <div
                        className="input-group-btn input-group-btn"
                      >
                        <Button
                          color="secondary"
                          tag="button"
                          type="submit"
                        >
                          <button
                            className="btn btn-secondary"
                            onClick={[Function]}
                            type="submit"
                          >
                            Submit
                          </button>
                        </Button>
                      </div>
                    </InputGroupButton>
                  </div>
                </InputGroup>
              </form>
            </Form>
          </div>
        </div>
      </div>
      <div
        className="sticky-bar text-center"
      >
        <a
          className="nav-link d-inline"
        >
          <i
            className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
          />
        </a>
        <a
          className="nav-link d-inline"
          title="New Recipe"
        >
          <FontAwesome
            id="add-edit-modal-icon"
            name="plus"
            onClick={[Function]}
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-plus fa-lg"
              id="add-edit-modal-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
        >
          <FontAwesome
            className="search-icon"
            name="search"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-search fa-lg search-icon"
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
        >
          <FontAwesome
            name="heart"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-heart fa-lg"
            />
          </FontAwesome>
        </a>
      </div>
    </div>
  </div>
</StickyBar>
`;

exports[`StickyBar renders share icon when type 'home' is passed 1`] = `
<StickyBar
  currentLocation="home"
  dispatch={[Function]}
>
  <div
    className="home"
  >
    <div
      className="sticky-bar-wrapper text-center d-block d-sm-block d-md-none"
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
          id="result-search-form-div"
        >
          <div
            className="col-12 w-100 py-3"
          >
            <Form
              id="search-form"
              tag="form"
            >
              <form
                className=""
                id="search-form"
              >
                <InputGroup
                  tag="div"
                >
                  <div
                    className="input-group"
                  >
                    <Label
                      className="sr-only"
                      for="search"
                      tag="label"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <label
                        className="sr-only  form-control-label"
                        htmlFor="search"
                      >
                        Search for Recipe
                      </label>
                    </Label>
                    <Input
                      id="search"
                      placeholder="Search for More Recipes"
                      tag="p"
                      type="text"
                    >
                      <input
                        className="form-control"
                        id="search"
                        placeholder="Search for More Recipes"
                        type="text"
                      />
                    </Input>
                    <InputGroupButton
                      className="input-group-btn"
                      tag="div"
                    >
                      <div
                        className="input-group-btn input-group-btn"
                      >
                        <Button
                          color="secondary"
                          tag="button"
                          type="submit"
                        >
                          <button
                            className="btn btn-secondary"
                            onClick={[Function]}
                            type="submit"
                          >
                            Submit
                          </button>
                        </Button>
                      </div>
                    </InputGroupButton>
                  </div>
                </InputGroup>
              </form>
            </Form>
          </div>
        </div>
      </div>
      <div
        className="sticky-bar text-center"
      >
        <a
          className="nav-link d-inline"
          data-toggle="modal"
        >
          <FontAwesome
            id="social-modal-icon"
            name="share-alt"
            onClick={[Function]}
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-share-alt fa-lg"
              id="social-modal-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
          title="New Recipe"
        >
          <FontAwesome
            id="add-edit-modal-icon"
            name="plus"
            onClick={[Function]}
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-plus fa-lg"
              id="add-edit-modal-icon"
              onClick={[Function]}
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
        >
          <FontAwesome
            className="search-icon"
            name="search"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-search fa-lg search-icon"
            />
          </FontAwesome>
        </a>
        <a
          className="nav-link d-inline"
        >
          <FontAwesome
            name="heart"
            size="lg"
          >
            <span
              aria-hidden={true}
              className="fa fa-heart fa-lg"
            />
          </FontAwesome>
        </a>
      </div>
    </div>
  </div>
</StickyBar>
`;
